afl-fuzz.c

```diff
@@ -503,15 +513,20 @@ static void bind_to_free_cpu(void) {
 
   }
 
-  OKF("Found a free CPU core, binding to #%u.", i);
+  OKF("Found a free CPU core, try binding to #%u.", i);
 
   cpu_aff = i;
 
   CPU_ZERO(&c);
   CPU_SET(i, &c);
 
-  if (sched_setaffinity(0, sizeof(c), &c))
-    PFATAL("sched_setaffinity failed");
+  if (sched_setaffinity(0, sizeof(c), &c)) {
+    if (cpu_start == cpu_core_count)
```

Comment:
Can you please re-write this without `goto`? IMO it's pretty confusing, given the slightly different loops on lines 496 and 500. One suggestion might be to create a helper function e.g. `find_next_available_cpu`, which would be called from a `while` loop calling `sched_setaffinity` until it either succeeds or until all `cpu_start` have been tried.

Reply:
@Dor1s done

Reply:
The patch you've uploaded have a lot of code duplication. I'm sorry if my previous comment wasn't clear. I meant to move only the platform-dependent logic into that function, and use a `while` loop instead of `goto`. Inside the loop you would call that new function and then do the same existing code, including `sched_setaffinity` call